compiler/rustc_infer/src/infer/error_reporting/mod.rs

```diff
@@ -84,6 +84,31 @@ pub use need_type_info::TypeAnnotationNeeded;
 
 pub mod nice_region_error;
 
+/// A helper for building inference-related errors. The `typeck_results`
+/// field is only populated during an in-progress typeck.
+/// Get an instance by calling `InferCtxt::err` or `FnCtxt::infer_err`.
+pub struct InferErrCtxt<'a, 'tcx> {
```

Comment:
maybe `TypeErrCtxt` and rename `FnCtxt::infer_err` and `InferCtxt::err` to `fn (Infer|FnCtxt)::err_ctxt` :think

Reply:
SGTM